core/src/main/java/org/apache/accumulo/core/spi/compaction/CompactionPlan.java

```diff
@@ -55,5 +56,13 @@ Builder addJob(short priority, CompactionExecutorId executor,
     CompactionPlan build();
   }
 
+  /**
+   * Return the set of compaction candidates.
+   */
+  Set<CompactableFile> getCandidates();
```

Comment:
Thinking this should not be added to SPI.  It has nothing to do with the plan for a compaction, but is a subset of the information that went into creating a plan.   Also I suggested other changes that would not require this change.

Reply:
I was thinking since the `CompactionPlan` is the output of the `CompactionPlanner`, it would have the candidates in it. But now I see that `CompactionJob` has files as well. I will drop this from the plan. I am thinking now, it would be nice to be able to differentiate between the files as to when `CompactableFile` actually gets selected for compaction or not. Maybe we could do something similar to what I did with `TabletFile` and `StoredTabletFile`. 

Also, trying to write javadoc for `CompactableFile` made me wonder if we should make it a self contained collection since you need at least 2 files to compact them into 1. There shouldn't ever be a single CompactableFile by itself right?

Reply:
> There shouldn't ever be a single CompactableFile by itself right?

@EdColeman reminded me of the use case where a user configures the `AgeOffFilter` and runs a user compaction to drop old data. Do you know where this would occur in CompactableImpl? I can't find it in the new code.